styleguide/git/commit-messages.md

```diff
@@ -0,0 +1,38 @@
+# Commit Messages
+
+The commit messages must follow the rules in this document.
+
+As we build automated Change Logs for some projects please make sure you only use the specified types and formats.
+The scope may be optionally set.
+Note: the software creating the versions may not bump the version if there is neither a feature nor a fix since the last release. Please read the documentation form the used software for further information.
+
+
+The types are based on [Sparkbox's article](https://seesparkbox.com/foundry/semantic_commit_messages) about semantic commit messages.
+There is a [cli](https://github.com/fteem/git-semantic-commits) to use the commit messages with a simpler interface
+
+## Commit Message Format
+
+```
+feat(scope): add hat wobble
+^--^  ^------------^
+|     |
+|     +-> Summary in present tense.
+|
++-------> Type: chore, docs, feat, fix, refactor, style, or test.
+
+```
+- feat: mandatory, one of the listed types from below.
+- summary: mandatory, summary about your changes in present tense.
+- scope: optional, describes the component the code is changing.
+
+## Type
+
+Must be one of the following:
+
+- chore: A small change not affecting the functionality of the code.
+. docs: Changes or additions in the documentation.
+- feat: A feature.
+- fix: A bug fix.
+- refactor: A code change that neither fixes a bug nor adds a feature
+- style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
```

Comment:
```suggestion
- style: Changes that do not change the code (white-space, formatting, missing semi-colons, etc)
```

Reply:
I do not agree with that, as for example if you have legacy projects you will often update the syntax to ES6 wich does change the code but ist not affecting it's meaning.
What would be about "affecting the functionality of the code"?

Reply:
Isn't there the commit type `refactor` for this? An upgrade in es syntax is clearly a refactoring. Adding some whitespace isn't. 

`Style` type commits should not change a single letter of the code. It may for example add a newline inside a statement though: `do().something()` -> `do().\nsomething()`. 

The meaning of the word «meaning» is too broad. For example, if i have a function called `showButton` that adds a class on an html element, it won't change its meaning when it sets the style directly on the element in a new version. This is a `fix` or a `refactoring`, not a style change.